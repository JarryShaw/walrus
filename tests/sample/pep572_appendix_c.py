a = 42


def f():
    # `a` is local to `f`, but remains unbound
    # until the caller executes this genexp:
    yield ((a := i) for i in range(3))
    yield lambda: a + 100
    print('done')
    try:
        print('`a` is bound to {}'.format(a))
        assert False
    except UnboundLocalError:
        print('`a` is not yet bound')


results = list(f())
print(list(map(type, results)))
print(list(results[0]))
print(results[1]())
print(a)
