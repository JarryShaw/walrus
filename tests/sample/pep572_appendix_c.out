done
`a` is not yet bound
[<class 'generator'>, <class 'function'>]
[0, 1, 2]
102
42
